### PR TITLE
CE-2799 Make Flags API check permissions before any write action

### PR DESCRIPTION
**Pull request by adamkarminski**
I have also removed a few bad `try{} catch{}` structures.

/cc @lukaszkonieczny @Grunny

### CODE REVIEW

File: extensions/wikia/Flags/controllers/FlagsApiController.class.php [hunk 1]
```diff
@@ -132,22 +132,19 @@ public function getFlagsForPageForEdit() {
...
-			$flagModel = new Flag();
-			$modelResponse = $flagModel->addFlagsToPage( $this->params );
+		$flagModel = new Flag();
+		$modelResponse = $flagModel->addFlagsToPage( $this->params );
 
-			$this->getCache()->purgeFlagsForPage( $this->params['page_id'] );
-			$this->purgeFlaggedPages( $this->params['flags'] );
-			$this->purgeFlagInsights( $this->params['flags'] );
+		$this->getCache()->purgeFlagsForPage( $this->params['page_id'] );
+		$this->purgeFlaggedPages( $this->params['flags'] );
```
Comment (bkoval) [inline extensions/wikia/Flags/controllers/FlagsApiController.class.php]: I can see a couple of calls to $this->params['flags'], maybe it's result could be cached in a local var?

Comment (adamkarminski) [inline extensions/wikia/Flags/controllers/FlagsApiController.class.php]: I don't think it's a bad idea. :shipit:

**Comment by Grunny** [comment]
LGTM.

**Comment by adamkarminski** [comment]
ping @bkoval

**Comment by bkoval** [comment]
Beautiful!

**Pull request merged**